src/ompl/base/src/OptimizationObjective.cpp

```diff
@@ -78,6 +78,11 @@ bool ompl::base::OptimizationObjective::isCostEquivalentTo(Cost c1, Cost c2) con
     return !isCostBetterThan(c1, c2) && !isCostBetterThan(c2, c1);
 }
 
+bool ompl::base::OptimizationObjective::isCostEquivalentTo(Cost c1, Cost c2, double threshold) const
+{
+    return isCostEquivalentTo(c1, c2) || (std::fabs(c1.value() - c2.value()) < threshold);
+}
+
```

Comment:
I believe the second condition should be something like `isCostBetterThan(subtractCosts(c1, c2), threshold)` where `threshold` is a cost, similar to the existing `OptimizationObjective` threshold?